RealmSwift/Sync.swift

```diff
@@ -666,3 +666,378 @@ extension Results where Element == SyncPermission {
 /// :nodoc:
 @available(*, unavailable, renamed: "SyncPermission")
 public final class SyncPermissionValue { }
+
+// MARK: - Permissions
+
+/**
+ A permission which can be applied to a Realm, Class, or specific Object.
+
+ Permissions are applied by adding the permission to the RealmPermission singleton
+ object, the ClassPermission object for the desired class, or to a user-defined
+ List<Permission> property on a specific Object instance. The meaning of each of
+ the properties of Permission depend on what the permission is applied to, and so are
+ left undocumented here. See `RealmPrivileges`, `ClassPrivileges`, and
+ `ObjectPrivileges` for details about what each of the properties mean when applied to
+ that type.
+ */
+public class Permission: Object {
+    /// The Role which this Permission applies to. All users within the Role are
+    /// granted the permissions specified by the fields below any
+    /// objects/classes/realms which use this Permission.
+    ///
+    /// This property cannot be modified once set.
+    @objc dynamic public var role: PermissionRole?
+
+    @objc dynamic public var canRead = false
+    @objc dynamic public var canUpdate = false
+    @objc dynamic public var canDelete = false
+    @objc dynamic public var canSetPermissions = false
+    @objc dynamic public var canQuery = false
+    @objc dynamic public var canCreate = false
+    @objc dynamic public var canModifySchema = false
+
+    @objc override public class func _realmObjectName() -> String {
+        return "__Permission"
+    }
+}
+
+/**
+ A Role within the permissions system.
+
+ A Role consists of a name for the role and a list of users which are members of the role.
+ Roles are granted privileges on Realms, Classes and Objects, and in turn grant those
+ privileges to all users which are members of the role.
+
+ A role named "everyone" is automatically created in new Realms, and all new users which
+ connect to the Realm are automatically added to it. Any other roles you wish to use are
+ managed as normal Realm objects.
+ */
+public class PermissionRole: Object {
+    @objc dynamic public var name = ""
+    let users = List<PermissionUser>()
+
+    @objc override public class func _realmObjectName() -> String {
```

Comment:
`sh build.sh verify-docs` is complaining about the lack of documentation on these methods, too. I guess a `/// :nodoc:` would suffice to silence it.